src/vs/platform/shell/node/shellEnv.ts

```diff
@@ -116,18 +118,11 @@ async function doResolveShellEnv(logService: ILogService, token: CancellationTok
 	const systemShell = await getSystemShell(OS, env); // note: windows always resolves a powershell instance
 	logService.trace('doResolveShellEnv#shell', systemShell);
 
-	let name = basename(systemShell);
-	if (isWindows) {
-		const nameExt = extname(name);
-		if (getWindowPathExtensions().some(e => equalsIgnoreCase(e, nameExt))) {
-			name = name.substring(0, name.length - nameExt.length); // remove any .exe/.cmd/... from the name for matching logic on Windows
-		}
-	}
-
+	const name = basename(systemShell);
 	let command: string, shellArgs: Array<string>;
 	const extraArgs = '';
 	if (/^(?:pwsh|powershell)(?:-preview)?$/.test(name)) {
-		const profilePaths = await getPowershellProfilePaths(systemShell);
+		const profilePaths = await getPowershellProfilePaths();
 		const profilePathThatExists = await first(profilePaths.map(profilePath => async () => (await FSPromises.exists(profilePath)) ? profilePath : undefined));
```

Comment:
I don't think we need to be doing this anymore? The less awaits in this code the better.

Previously it looked like this:

```
			// Older versions of PowerShell removes double quotes sometimes so we use "double single quotes" which is how
			// you escape single quotes inside of a single quoted string.
			command = `& '${process.execPath}' ${extraArgs} -p '''${mark}'' + JSON.stringify(process.env) + ''${mark}'''`;
			shellArgs = ['-Login', '-Command'];
```